### PR TITLE
DiskImageBuilder remove old folder and files

### PR DESCRIPTION
**Pull request by dglynn**
Changed the image_functions.sh to look for the right directory which
is "$IMAGE_NAME.d" and remove this. Added a check to remove old *.qcow2
files from previous builds. Confirmed the instance and image cleanup is
deleting these using the uuid in the db and not by their name. Added the
IMAGE_VISIBILITY var to all image_create as this needs setting when we
make new images private when testing.i Fixed TEST_IMAGE_NAME var as it
was causing the script to fail, as there was an image and instance using
the same names. Added a redirect of STDERR as this was causing the
delete files in the cleanup function to fail.

### CODE REVIEW

File: scripts/image_functions.sh [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
-	# delete any orphaned temporary images
+	find . -name "${IMAGE_NAME}*.d" -type d -exec rm -rf {} +
+    find . -name "${IMAGE_NAME}*.qcow2" -type f -exec rm -f {} +
+	# delete any orphaned temporary images using image_id
 	for id in $(glance image-list | grep "$TMP_IMAGE_NAME" | awk '{print $2}')
 	do
 		glance image-delete $id &>/dev/null || true
 	done
-	# delete any orphaned test instance
+	# delete any orphaned test instance using instance_id
```
Comment (oscarkraemer) [inline scripts/image_functions.sh]: instane_id ? or instance_name?

File: scripts/image_functions.sh [hunk 2]
```diff
@@ -2,13 +2,14 @@
 export PYTHONIOENCODING
 
 function cleanup() {
-	find . -name "${IMAGE_NAME}" -type d -exec rm -rf '{}' \;
-	# delete any orphaned temporary images
+	find . -name "${IMAGE_NAME}*.d" -type d -exec rm -rf {} +
+    find . -name "${IMAGE_NAME}*.qcow2" -type f -exec rm -f {} +
```
Comment (oscarkraemer) [inline scripts/image_functions.sh]: Indentation seems of. Urk, I reviewing 'rm -f' commands... ...so much power so much that can go wrong, but it seems okay.

**Review by oscarkraemer — COMMENTED** [review]
Otherwise it looks okay

File: scripts/image_functions.sh (hunk 1 above)
Comment (dglynn) [reply scripts/image_functions.sh]: Instance_id, testing a new commit of this branch

Comment (oscarkraemer) [reply scripts/image_functions.sh]: Woosp I meant on row 13. 'for id in $(nova list | grep "$TEST_INSTANCE_NAME" | awk '{print $2}')'

File: scripts/image_functions.sh (hunk 2 above)
Comment (dglynn) [reply scripts/image_functions.sh]: this is just adding to what was there before. the indentation seems ok in the file when copied down, weird

**Pull request closed without merging**